lib/natalie/compiler/pass0c.rb

```diff
@@ -24,7 +25,62 @@ def process_iter(exp)
         )
       end
 
+      def process_defined(exp, block_with_args=nil)
+        @in_defined_statement = true
+        exp = exp.new(*exp.map {|sub_exp| sub_exp.is_a?(Sexp) ? process(sub_exp) : sub_exp})
+        @in_defined_statement = false
+        return exp
+      end
+
+      def type_is_always_truthy(arg)
+        # FIXME: Are there any other types that are always truthy?
+        return [:true,:str, :lit, :array].include?(arg)
```

Comment:
missing space nitpick:

```suggestion
        return [:true, :str, :lit, :array].include?(arg)
```